src/lib/filters/comp_filter.cpp

```diff
@@ -25,7 +25,12 @@ Compression_Decompression_Filter::Compression_Decompression_Filter(Transform* tr
    {
    m_transform.reset(dynamic_cast<Compressor_Transform*>(transform));
    if(!m_transform)
-      throw std::invalid_argument("Transform " + transform->name() + " is not a compressor");
+      {
+      if (transform)
+         throw std::invalid_argument("Transform " + transform->name() + " is not a compressor");
+      else
+         throw std::invalid_argument("Transform is null");
+      }
```

Comment:
Could you move this part into a separate PR please? This might get in more easily since it is quickly reviewed.